internal/worker.go

```diff
@@ -271,6 +271,12 @@ type (
 		//
 		// Deprecated: All bugports are always deprecated and may be removed at any time.
 		WorkerBugPorts WorkerBugPorts
+
+		// Optional: This implementation ensures that a specific function is executed only once per instance.
+		// The mechanism can be overridden by other interfaces that implement the 'Do()' method.
+		//
+		// default: nil, that would ensure some functions are executed only once
+		Sync oncePerHost
```

Comment:
This is user visible worker options. we shouldn't expose `oncePerHost` here.

Reply:
If we do not exposed that here, we might need to insert that as part of the `workerExecutionParameters` and add the Sync.Once as part of the parameter into function "NewWorker" and "newAggregatedWorker". What do you think about that idea?

Reply:
it makes sense to add to `workerExecutionParameters` as it is not exposed outside. I'd also recommend looking at `WithSomeOption` pattern in go https://www.sohamkamani.com/golang/options-pattern/.